test/integration/targets/ansible-doc/filter_plugins/other.py

```diff
@@ -0,0 +1,22 @@
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.utils.display import Display
+
+display = Display()
+
+
+def donothing(a):
+    return a
+
+
+class FilterModule(object):
+    ''' Ansible core jinja2 filters '''
+
+    def filters(self):
+        return {
+            'donothing': donothing,
```

Comment:
It would be nice to have a test with a legacy plugin with the same name as a builtin too. It doesn't look like it's possible to list legacy plugins that shadow a builtin name or show their documentation/lack thereof.

I tested with a filter named `split` in ./filter_plugins and tested with:
`ansible-doc ansible.legacy --list -t filter -vvv --playbook-dir ./` and `ansible-doc ansible.legacy.split --playbook-dir ./ -vvv` (after adding docs for builtin to work around the traceback) and only the builtin info is shown.